connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -3629,6 +3630,64 @@ public void testPollDurationOnSlowConnectorOperations() {
         PowerMock.verifyAll();
     }
 
+    @Test(expected = RejectedExecutionException.class)
+    @SuppressWarnings("unchecked")
+    public void shouldThrowWhenStartAndStopExecutorThrowsRejectedExecutionExceptionAndHerderNotStopping() throws InterruptedException {
+        ExecutorService startAndStopExecutor = EasyMock.mock(ExecutorService.class);
+        herder.startAndStopExecutor = startAndStopExecutor;
+
+        Callable<Void> connectorStartingCallable = () -> null;
+
+        EasyMock.expect(startAndStopExecutor.invokeAll(EasyMock.anyObject(Collection.class))).andThrow(new RejectedExecutionException());
+
+        PowerMock.replayAll(startAndStopExecutor);
+
+        herder.startAndStop(Collections.singletonList(connectorStartingCallable));
+
+    }
+
+    @Test
+    public void shouldHaltCleanlyWhenHerderStartsAndStopsAndConfigTopicReadTimesOut() throws TimeoutException {
```

Comment:
This isn't really testing what it claims to be testing. What I'd expect from at test like this is verification that even if the herder's tick thread is blocked during a read of the config topic, `Herder::stop` can and will proceed gracefully after any graceful shutdown timeouts have expired.

In reality, what's happening is that `Herder::tick` is called once, returns immediately (gracefully handling a `TimeoutException` while attempting to read to the end of the config topic), and then `Herder::stop` is called afterward without any 
other simultaneous herder activity (like a blocked read to the end of the config topic) taking place.

TBH, although it'd be nice to have coverage for a case like this, I don't think it's strictly necessary. We could theoretically use the `startBackgroundHerder` utility method to have `tick` continuously called in the background, set up a block while reading the config topic that never returns (at least until the rest of the test is complete), and then invoke `Herder::stop` on the main test thread, but that would either require heavy mocking out of the herder's executors (which would diminish the value of the test) or take a long time (i.e., the entire herder graceful shutdown timeout) for each test run to complete. 

Neither of these options really seem worth it and I trust the changes in the main parts of the code base enough that I'm okay without adding additional coverage for them this time around.

If that seems reasonable, we can just remove this test case.

Reply:
yeah I was not totally sure on how to keep invoking the tick `method` repeatedly and hence ended up doing something like this. I also agree that it's really not too representative of what needs to be tested. I guess removing the test might be a better option.